### PR TITLE
Changed logic for setting collapsed property to true

### PR DESCRIPTION
**Pull request by YajurG**
## Description
Changed logic in FrameworkElementViewManager for setting 'collapsed' accessibility state so that collapsed is not set to true by default when expanded is false.

### Type of Change

- Bug fix (non-breaking change which fixes an issue)

### Why

Resolves [#11734]

### What
Checks to see if the expanded property is null i.e not set by the component. If so, sets collapsed to false instead of defaulting to true when expanded is false. If property is not null, then performs check and sets collapsed property based on value.

Tested locally on RNTester Button page in playground. Seems to be performing as desired there - Narrator no longer making 'collapsed' announcement on buttons and UIA tree reflects this with the ExpandCollapsed property being of LeafNode instead of collapsed. Tested on Buttons and Touchable components in the APIs/Accessibility section in RNTester, and they performed as expected with respect to accessibilityState and the 'expanded' prop, and removed announcement of 'collapsed' on buttons.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/11756)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/Views/FrameworkElementViewManager.cpp [hunk 1]
```diff
@@ -493,7 +493,9 @@ bool FrameworkElementViewManager::UpdateProperty(
             states[static_cast<int32_t>(winrt::Microsoft::ReactNative::AccessibilityStates::Expanded)] =
                 !innerValue.IsNull() && innerValue.AsBoolean();
             states[static_cast<int32_t>(winrt::Microsoft::ReactNative::AccessibilityStates::Collapsed)] =
-                innerValue.IsNull() || !innerValue.AsBoolean();
+                innerValue.IsNull() ? false : !innerValue.AsBoolean();
+            // states[static_cast<int32_t>(winrt::Microsoft::ReactNative::AccessibilityStates::Collapsed)] =
```
Comment (TatianaKapos) [inline vnext/Microsoft.ReactNative/Views/FrameworkElementViewManager.cpp:497]: Did you mean to leave this comment in?

File: vnext/Microsoft.ReactNative/Views/FrameworkElementViewManager.cpp [hunk 2]
```diff
@@ -493,7 +493,9 @@ bool FrameworkElementViewManager::UpdateProperty(
             states[static_cast<int32_t>(winrt::Microsoft::ReactNative::AccessibilityStates::Expanded)] =
                 !innerValue.IsNull() && innerValue.AsBoolean();
```
Comment (jonthysell) [inline vnext/Microsoft.ReactNative/Views/FrameworkElementViewManager.cpp:494]: I recommend extracting the logic here for getting the value for 'expanded' as a boolean variable, then setting expanded state to that value and collapsed to the opposite.

File: vnext/Microsoft.ReactNative/Views/FrameworkElementViewManager.cpp [hunk 3]
```diff
@@ -493,7 +493,9 @@ bool FrameworkElementViewManager::UpdateProperty(
...
             states[static_cast<int32_t>(winrt::Microsoft::ReactNative::AccessibilityStates::Collapsed)] =
-                innerValue.IsNull() || !innerValue.AsBoolean();
+                innerValue.IsNull() ? false : !innerValue.AsBoolean();
```
Comment (jonthysell) [inline vnext/Microsoft.ReactNative/Views/FrameworkElementViewManager.cpp:496]: Does this fix the behavior in the bug? if `innerName == "expanded"` that means expanded is present in the object, so collapsed should be set to false if expanded is true? Is someone sets `expanded: null` that's not really a supported scenario.

File: vnext/Microsoft.ReactNative/Views/FrameworkElementViewManager.cpp (hunk 1 above)
Comment (YajurG) [reply vnext/Microsoft.ReactNative/Views/FrameworkElementViewManager.cpp:497]: I left it in just as a reference for myself for what was there before - will remove before I merge it in

File: vnext/Microsoft.ReactNative/Views/FrameworkElementViewManager.cpp (hunk 3 above)
Comment (YajurG) [reply vnext/Microsoft.ReactNative/Views/FrameworkElementViewManager.cpp:496]: If `expanded: null`, then wouldn't the `innerValue.isNull()` evaluate to true, which would mean that collapsed would be false? My logic for the change was that if `expanded: null`, then it's not been set as a property, which means it shouldn't be expanded or collapsed, which in that case both expanded and collapsed would be set to false.

Comment (jonthysell) [reply vnext/Microsoft.ReactNative/Views/FrameworkElementViewManager.cpp:496]: Yeah, but the API is: expanded is either present with true/false or completely absent, not ever supposed to be set to null.

If your logic fixes the narrator reporting that's good, but there's a greater issue that UIA shouldn't probably see that the element could be expanded/collapsed pattern **at all** if expanded isn't set. Which can be a separate bug.

**Review by jonthysell — APPROVED** [review]

**Review by TatianaKapos — APPROVED** [review]

**Pull request merged**